components/AdditionalDatasheetEditor.tsx

```diff
@@ -0,0 +1,295 @@
+'use client';
+
+import { useCallback, useEffect, useState } from 'react';
+import { useQuery, useMutation } from '@apollo/client';
+import {
+  Accordion as MuiAccordion,
+  AccordionDetails as MuiAccordionDetails,
+  AccordionSummary as MuiAccordionSummary,
+  Box,
+  Typography,
+  SxProps,
+  Theme,
+  Skeleton,
+} from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { DataGrid, GridColDef } from '@mui/x-data-grid';
+import { ChevronDown } from 'react-bootstrap-icons';
+
+import { GET_MEASURE_TEMPLATES } from '@/queries/get-measure-templates';
+import { UPDATE_MEASURE_DATAPOINT } from '@/queries/update-measure-datapoint';
+import { useFrameworkInstanceStore } from '@/store/selected-framework-instance';
+import { useSnackbar } from './SnackbarProvider';
+import { additionalMeasures } from '@/constants/measure-overrides';
+import CustomEditComponent from './DatasheetEditor';
+import { CustomFooter } from './DatasheetEditor';
+
+const Accordion = styled(MuiAccordion)(({ theme }) => ({
+  '&:first-of-type': {
+    borderTopLeftRadius: theme.shape.borderRadius,
+    borderTopRightRadius: theme.shape.borderRadius,
+  },
+  '&:last-of-type': {
+    borderBottomLeftRadius: theme.shape.borderRadius,
+    borderBottomRightRadius: theme.shape.borderRadius,
+  },
+  border: `1px solid ${theme.palette.divider}`,
+  '&:not(:last-child)': {
+    borderBottom: 0,
+  },
+  '&::before': {
+    display: 'none',
+  },
+}));
+
+const DATA_GRID_SX: SxProps<Theme> = (theme) => ({
+  '& .MuiDataGrid-columnHeaderTitle, & .MuiDataGrid-cell': {
+    display: 'flex',
+    fontSize: theme.typography.caption.fontSize,
+    whiteSpace: 'normal',
+    lineHeight: 'normal',
+    py: 1,
+    justifyContent: 'flex-start',
+    textAlign: 'left',
+  },
+  '& .MuiDataGrid-cell': {
+    textAlign: 'left',
+  },
+});
+
+interface KeyMeasure {
+  label: string;
+  question: string;
+}
+
+interface MeasureDataPoint {
+  id: string;
+  question: string;
+  baselineYear: string | number;
+  unit: string;
+  [year: number]: string | number;
+}
+
+type MeasureSection = Record<string, MeasureDataPoint[]>;
+
+export function AdditionalDatasheetEditor() {
+  const baselineYear =
+    useFrameworkInstanceStore((state) => state.baselineYear) ?? 0;
+  const selectedInstanceId = useFrameworkInstanceStore(
+    (state) => state.selectedInstance
+  );
+
+  const { setNotification } = useSnackbar();
+  const [measureSection, setMeasureSection] = useState<MeasureSection>({});
+  const [expanded, setExpanded] = useState<number | null>(0);
+
+  const { data, loading, error } = useQuery(GET_MEASURE_TEMPLATES, {
+    variables: { frameworkConfigId: selectedInstanceId },
+  });
+
+  const currentYear = new Date().getFullYear();
+  const additionalYears = Array.from(
+    { length: currentYear - baselineYear - 1 },
+    (_, i) => baselineYear + 1 + i
+  );
+
+  const [updateMeasureDataPoint] = useMutation(UPDATE_MEASURE_DATAPOINT);
```

Comment:
Can you add the same loading logic as in the `DatasheetEditor` [here](https://github.com/kausaltech/nzc-data-studio/blob/e72854fd24eae55f9f9fa89fa3c9aac1bcdee5fb/components/DatasheetEditor.tsx#L702) and [here](https://github.com/kausaltech/nzc-data-studio/blob/e72854fd24eae55f9f9fa89fa3c9aac1bcdee5fb/components/DatasheetEditor.tsx#L816)?